File: src/com/lysdev/transperthcached/livetimes/TimesForStation.java
package com.lysdev.transperthcached.livetimes;

import java.util.List;
import org.joda.time.DateTime;


public class TimesForStation {
    private String station_name;
    private DateTime last_updated;
    private List<Trip> trips;

    public TimesForStation(String station_name, DateTime last_updated, List<Trip> trips) {
        this.station_name = station_name;
        this.last_updated = last_updated;
        this.trips = trips;
    }

    public List<Trip> getTrips() {
        return this.trips;
    }

    public String toString() {
        return String.format(
            "<TimesForStation %s - %s - %d trips>",
            this.station_name,
            this.last_updated.toString(),
            this.trips.size()
        );
    }
}

File: src/com/lysdev/transperthcached/livetimes/TimesForStation.scala
package com.lysdev.transperthcached.livetimes;

import org.joda.time.DateTime;


class TimesForStation(
        station_name: String,
        last_updated: DateTime,
        trips: List[Trip]) {

    def getTrips()       = this.trips
    def getLastUpdated() = this.last_updated
    def getStationName() = this.station_name

    override def toString() = "<TimesForStation $getStationName - $getLastUpdated - ${getTrips.size} trips>"
}
